packages/amplify-cli/src/extensions/amplify-helpers/open-editor.ts

```diff
@@ -19,47 +21,89 @@ export async function openEditor(context, filePath, waitToContinue = true) {
 
   if (editorSelected !== 'none') {
     const editorArguments: string[] = [];
-    const editor = envEditor.getEditor(editorSelected);
+
+    let editor: envEditor.Editor;
+
+    // Until env-editor supports xcode we've to have a branch here
+    if (editorSelected === 'xcode') {
+      editor = {
+        id: 'xcode',
+        name: 'Xcode',
+        binary: 'xcode',
+        isTerminalEditor: false,
+        paths: ['/Applications/Xcode.app/Contents/MacOS/Xcode', '/Applications/Xcode-beta.app/Contents/MacOS/Xcode'],
```

Comment:
this seems brittle. Will these paths always be valid? (or valid on the vast majority of macs)?

Reply:
We use `env-editor` package, and this is the object we get and complies with how it detects the editors, hence it is here. Will submit a PR for that package as I did for Android Studio support as well. Yes the apps are there like this.